runtime/syntax/unison.vim

```diff
@@ -0,0 +1,122 @@
+syn match   uCharacter		"^'\([^\\]\|\\[^']\+\|\\'\)'" contains=uSpecialChar,uSpecialCharError
+syn match   uNumber		"\<[0-9]\+\>\|\<0[xX][0-9a-fA-F]\+\>\|\<0[oO][0-7]\+\>"
+syn match   uFloat		"\<[0-9]\+\.[0-9]\+\([eE][-+]\=[0-9]\+\)\=\>"
+
+" Keyword definitions. These must be patterns instead of keywords
+" because otherwise they would match as keywords at the start of a
+" "literate" comment (see lu.vim).
+syn match uModule		"\<namespace\>"
+syn match uImport		"\<use\>"
+syn match uTypedef		"\<\(unique\|structural\|∀\|forall\)\>"
+syn match uStatement		"\<\(ability\|do\|type\|where\|match\|cases\|;\|let\|with\|handle\)\>"
+syn match uConditional		"\<\(if\|else\|then\)\>"
+
+syn match uBoolean "\<\(true\|false\)\>"
+
+syn match uType "\<\C[A-Z][0-9A-Za-z_'!]*\>"
+syn match uName "\<\C[a-z_][0-9A-Za-z_'!]*\>"
+
+" Comments
+syn match   uLineComment      "---*\([^-!#$%&\*\+./<=>\?@\\^|~].*\)\?$"
+syn region  uBlockComment     start="{-"  end="-}" contains=uBlockComment
+syn region  uPragma	      start="{-#" end="#-}"
+syn region  uBelowFold	      start="^---" skip="." end="." contains=uBelowFold
+
+" Docs
+syn region  uDocBlock         matchgroup=unisonDoc start="{{" end="}}" contains=uDocTypecheck,uDocQuasiquote,uDocDirective,uDocCode,uDocCodeInline,uDocCodeRaw,uDocMono,@markdownLikeDocs
+syn region  uDocQuasiquote    contained matchgroup=unisonDocQuote start="{{" end= "}}" contains=TOP
+syn region  uDocCode          contained matchgroup=unisonDocCode start="^\s*```\s*$" end="^\s*```\s*$" contains=TOP
+syn region  uDocTypecheck     contained matchgroup=unisonDocCode start="^\s*@typecheck\s*```\s*$" end="^\s*```\s*$" contains=TOP
+syn region  uDocCodeRaw       contained matchgroup=unisonDocCode start="^\s*```\s*raw\s*$" end="^\s*```\s*$" contains=NoSyntax
+syn region  uDocCodeInline    contained matchgroup=unisonDocCode start="`\@<!``" end="`\@<!``" contains=TOP
+syn match   uDocMono          "''[^']*''"
+syn region  uDocDirective     contained matchgroup=unisonDocDirective start="\(@\([a-zA-Z0-9_']*\)\)\?{{\@!" end="}" contains=TOP
+
+syn match uDebug "\<\(todo\|bug\|Debug.trace\)\>"
+
+" things like 
+"    > my_func 1 3
+"    test> Function.tap.tests.t1 = check let
+"      use Nat == +
+"      ( 99, 100 ) === (withInitialValue 0 do
+"          :      :      :
+syn match uWatch "^[A-Za-z]*>"
+
+" Define the default highlighting.
+" For version 5.7 and earlier: only when not done already
+" For version 5.8 and later: only when an item doesn't have highlighting yet
+if version >= 508 || !exists("did_u_syntax_inits")
+  if version < 508
+    let did_u_syntax_inits = 1
+    command -nargs=+ HiLink hi link <args>
+  else
+    command -nargs=+ HiLink hi def link <args>
+  endif
+   
+   HiLink       uWatch                           Debug
+   HiLink       uDocMono                         Delimiter
+   HiLink       unisonDocDirective               Import
```

Comment:
There's a mix of prefixes for the group names.  The prefix should match the filetype name of "unison".